File: email-backup.java/src/test/java/demo/FlagsTest.java
package demo;

import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertNotNull;

public class FlagsTest {

    private static final String EMAIL = "email";
    private static final String PASSWORD = "password";
    private static final String HOST = "host";
    private static final int PORT = 465;
    private static final String EXTENSION = "extension";
    private static final String FILE = "file";
    private static final String EMAIL2 = "email2";

    @Test
    public void constructorTest1() {
        Flags flags = new Flags(EMAIL, PASSWORD, HOST, PORT, null, null, null);

        assertEquals(flags.getEmail(), EMAIL);
        assertEquals(flags.getPassword(), PASSWORD);
        assertEquals(flags.getHost(), HOST);
        assertEquals(flags.getPort(), PORT);

        assertNotNull(flags.getExcludedExtensions());
        assertEquals(flags.getExcludedExtensions().size(), 0);

        assertNotNull(flags.getIncludedFiles());
        assertEquals(flags.getIncludedFiles().size(), 0);

        assertNotNull(flags.getEmailsTo());
        assertEquals(flags.getEmailsTo().size(), 0);
    }

    @Test
    public void constructorTest2() {
        Flags flags = new Flags(EMAIL, PASSWORD, HOST, PORT, new String[]{EXTENSION}, new String[]{FILE}, new String[]{EMAIL2});

        assertEquals(flags.getEmail(), EMAIL);
        assertEquals(flags.getPassword(), PASSWORD);
        assertEquals(flags.getHost(), HOST);
        assertEquals(flags.getPort(), PORT);

        assertNotNull(flags.getExcludedExtensions());
        assertEquals(flags.getExcludedExtensions().size(), 1);
        assertEquals(flags.getExcludedExtensions().get(0), EXTENSION);

        assertNotNull(flags.getIncludedFiles());
        assertEquals(flags.getIncludedFiles().size(), 1);
        assertEquals(flags.getIncludedFiles().get(0), FILE);

        assertNotNull(flags.getEmailsTo());
        assertEquals(flags.getEmailsTo().size(), 1);
        assertEquals(flags.getEmailsTo().get(0), EMAIL2);
    }
}

File: email-backup.kotlin/src/test/kotlin/demo/FlagsTest.kt
package demo

import org.testng.Assert.assertEquals
import org.testng.Assert.assertNotNull
import org.testng.annotations.Test

public class FlagsTest {

    private val EMAIL = "email"
    private val PASSWORD = "password"
    private val HOST = "host"
    private val PORT = 465
    private val EXTENSION = "extension"
    private val FILE = "file"
    private val EMAIL2 = "email2"

    @Test
    public fun constructorTest1() {
        val flags = Flags(EMAIL, PASSWORD, HOST, PORT, null, null, null)

        assertEquals(flags.email, EMAIL)
        assertEquals(flags.password, PASSWORD)
        assertEquals(flags.host, HOST)
        assertEquals(flags.port, PORT)

        assertNotNull(flags.excludedExtensions)
        assertEquals(flags.excludedExtensions.size(), 0)

        assertNotNull(flags.includedFiles)
        assertEquals(flags.includedFiles.size(), 0)

        assertNotNull(flags.emailsTo)
        assertEquals(flags.emailsTo.size(), 0)
    }

    @Test
    public fun constructorTest2() {
        val flags = Flags(EMAIL, PASSWORD, HOST, PORT, arrayOf(EXTENSION), arrayOf(FILE), arrayOf(EMAIL2))

        assertEquals(flags.email, EMAIL)
        assertEquals(flags.password, PASSWORD)
        assertEquals(flags.host, HOST)
        assertEquals(flags.port, PORT)

        assertNotNull(flags.excludedExtensions)
        assertEquals(flags.excludedExtensions.size(), 1)
        assertEquals(flags.excludedExtensions.get(0), EXTENSION)

        assertNotNull(flags.includedFiles)
        assertEquals(flags.includedFiles.size(), 1)
        assertEquals(flags.includedFiles.get(0), FILE)

        assertNotNull(flags.emailsTo)
        assertEquals(flags.emailsTo.size(), 1)
        assertEquals(flags.emailsTo.get(0), EMAIL2)
    }
}
